plugins/image-prioritizer/helper.php

```diff
@@ -77,6 +77,142 @@ function image_prioritizer_register_tag_visitors( OD_Tag_Visitor_Registry $regis
 	$registry->register( 'image-prioritizer/video', $video_visitor );
 }
 
+/**
+ * Gets the script to lazy-load videos.
+ *
+ * Load a video and its poster image when it approaches the viewport using an IntersectionObserver.
+ *
+ * Handles 'autoplay' and 'preload' attributes accordingly.
+ *
+ * @since 0.2.0
+ */
+function image_prioritizer_get_lazy_load_script(): string {
+	$path   = image_prioritizer_get_asset_path( 'lazy-load.js' );
+	$script = file_get_contents( __DIR__ . '/' . $path ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
```

Comment:
Any advantage or disadvantage to using `WP_Filesystem_Direct` for local file access?

Reply:
Actually, `WP_Filesystem_Direct` uses the same:

https://github.com/WordPress/wordpress-develop/blob/daad8631f707f730530711ed874efe94c7fec968/src/wp-admin/includes/class-wp-filesystem-direct.php#L30-L40

And WordPress core uses `file_get_contents()` itself in similar situations:

* https://github.com/WordPress/wordpress-develop/blob/daad8631f707f730530711ed874efe94c7fec968/src/wp-includes/class-wp-editor.php#L527
* https://github.com/WordPress/wordpress-develop/blob/daad8631f707f730530711ed874efe94c7fec968/src/wp-includes/embed.php#L520-L522
* https://github.com/WordPress/wordpress-develop/blob/daad8631f707f730530711ed874efe94c7fec968/src/wp-includes/embed.php#L1088
* https://github.com/WordPress/wordpress-develop/blob/daad8631f707f730530711ed874efe94c7fec968/src/wp-includes/formatting.php#L5947